app/models/stats.rb

```diff
@@ -0,0 +1,120 @@
+# typed: false
+
+class Stats
+  FIRST_MONTH = Time.new(2012, 7, 3).utc.freeze
+  TIMESCALE_DIVISIONS = "1 year".freeze
+  USERS_GRAPH_CACHE_KEY = "users_graph".freeze
+  ACTIVE_USERS_GRAPH_CACHE_KEY = "active_users_graph".freeze
+  STORIES_GRAPH_CACHE_KEY = "stories_graph".freeze
+  COMMENTS_GRAPH_CACHE_KEY = "comments_graph".freeze
+  VOTES_GRAPH_CACHE_KEY = "votes_graph".freeze
+
+  def self.get_users_graph = get_cached_graph(USERS_GRAPH_CACHE_KEY)
+
+  def self.get_active_users_graph = get_cached_graph(ACTIVE_USERS_GRAPH_CACHE_KEY)
+
+  def self.get_stories_graph = get_cached_graph(ACTIVE_USERS_GRAPH_CACHE_KEY)
+
+  def self.get_comments_graph = get_cached_graph(COMMENTS_GRAPH_CACHE_KEY)
+
+  def self.get_votes_graph = get_cached_graph(VOTES_GRAPH_CACHE_KEY)
+
+  def self.fill_users_graph_cache
+    cache_monthly_graph(USERS_GRAPH_CACHE_KEY, {
+      graph_title: "Users joining by month",
+      scale_y_divisions: 100,
+      read_only: read_only
+    }) {
+      User.pluck(:created_at).map { |created_at| created_at.strftime("%Y-%m") }.tally.sort.flatten
+    }
+  end
+
+  def self.fill_active_users_graph_cache
+    monthly_graph(ACTIVE_USERS_GRAPH_CACHE_KEY, {
+      graph_title: "Active users by month",
+      scale_y_divisions: 500,
+      read_only: read_only
+    }) {
+      stories = Story.pluck(:created_at, :user_id).map { |created_at, user_id| [created_at.strftime("%Y-%m"), user_id] }
+      votes = Vote.pluck(:updated_at, :user_id).map { |updated_at, user_id| [updated_at.strftime("%Y-%m"), user_id] }
+      comments = Comment.pluck(:created_at, :user_id).map { |created_at, user_id| [created_at.strftime("%Y-%m"), user_id] }
+      combined = (stories + votes + comments).group_by(&:first).transform_values { |record| record.map(&:last).uniq.count }
+      combined.sort.flatten
+    }
+  end
+
+  def self.fill_stories_graph_cache
+    monthly_graph("stories_graph", {
+      graph_title: "Stories submitted by month",
+      scale_y_divisions: 250,
+      read_only: read_only
+    }) {
+      Story.pluck(:created_at).map { |created_at| created_at.strftime("%Y-%m") }.tally.sort.flatten
+    }
+  end
+
+  def self.fill_comments_graph_cache
+    monthly_graph("comments_graph", {
+      graph_title: "Comments posted by month",
+      scale_y_divisions: 1_000,
+      read_only: read_only
+    }) {
+      Comment.pluck(:created_at).map { |created_at| created_at.strftime("%Y-%m") }.tally.sort.flatten
+    }
+  end
+
+  def self.fill_votes_graph_cache
+    monthly_graph("votes_graph", {
+      graph_title: "Votes cast by month",
+      scale_y_divisions: 10_000,
+      read_only: read_only
+    }) {
+      Vote.pluck(:updated_at).map { |updated_at| updated_at.strftime("%Y-%m") }.tally.sort.flatten
+    }
+  end
+
+  def self.get_cached_graph(cache_key)
+    cached_data = Rails.cache.read(cache_key)
+    cached_data || "<i>No data</i>"
```

Comment:
```suggestion
    Rails.cache.read(cache_key) || "<i>No data</i>"
```

Don't need the intermediate variable.